### PR TITLE
[3.x] Fix refresh() on EmbedsOne

### LINKED ISSUE #1995: Model can't be refreshed when using EmbedsOne
- Laravel-mongodb Version: master
- PHP Version: 7.2 (Docker)
- Database Driver & Version: Same as Docker

### Description:
A Model can't receive `refresh()` when it has a EmbedsOne data, triggering an error on Eloquent Builder.

The bug don't happen when the EmbedsOne is not filled.

I'm already inspecting this, but the bug is inside the Eloquent Builder at all. :(

### Steps to reproduce
image

### Expected behaviour
Do the complete refresh()

### Actual behaviour
TypeError exception, when filled: second test

    TypeError : Argument 2 passed to Jenssegers\Mongodb\Relations\EmbedsOneOrMany::match() must be an instance of Illuminate\Database\Eloquent\Collection, null given, called in /code/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php on line 585
    /code/src/Jenssegers/Mongodb/Relations/EmbedsOneOrMany.php:78
    /code/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php:585
    /code/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php:554
    /code/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php:491
    /code/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php:1155
    /code/tests/EmbeddedRelationsTest.php:633

### PR DESCRIPTION
**Pull request by stephandesouza**
Fixes #1995 

Model::refresh() throws TypeError exception when using EmbedOne

### CODE REVIEW

File: src/Jenssegers/Mongodb/Relations/EmbedsOne.php [hunk 1]
```diff
@@ -28,6 +28,17 @@ public function getResults()
         return $this->toModel($this->getEmbedded());
     }
 
+    /**
+     * @inheritdoc
```
Comment (Smolevich) [inline src/Jenssegers/Mongodb/Relations/EmbedsOne.php]: Delete this doc block, please

Comment (stephandesouza) [reply src/Jenssegers/Mongodb/Relations/EmbedsOne.php]: Erm.. may I ask why?

Comment (Smolevich) [reply src/Jenssegers/Mongodb/Relations/EmbedsOne.php]: It is useless doc block

Comment (stephandesouza) [reply src/Jenssegers/Mongodb/Relations/EmbedsOne.php]: Well.. `inheritdoc` is better then no block, considering this method is a override. 

Proceed with the removal?

Comment (Smolevich) [reply src/Jenssegers/Mongodb/Relations/EmbedsOne.php]: Better type hints and doc blocks for params and return type  if type hints don't exist. Doc block `inheritdoc` seems like default useless doc block

Comment (stephandesouza) [reply src/Jenssegers/Mongodb/Relations/EmbedsOne.php]: Again why redoc when this is an override? This is not the default block `getEager()` is a method from Eloquent Builder.

Comment (divine) [reply src/Jenssegers/Mongodb/Relations/EmbedsOne.php]: I think it's better to keep it as it is.

Comment (Smolevich) [reply src/Jenssegers/Mongodb/Relations/EmbedsOne.php]: Using `inheritdoc` don't contain benefits here

**Pull request merged**